evm/rpc/server.py

```diff
@@ -0,0 +1,176 @@
+            raise HTTPMethodNotAllowed(method, self.allowed_methods)
+
+        block_dict = self._block_to_dict(block)
+        response = {"jsonrpc": "2.0", "id": req_id, "result": block_dict}
+        return web.json_response(response)
+
+    @asyncio.coroutine
+    def connect_peer(self, app):
+        return self.chain.on_demand_data_backend.get_peer()
+
+    @asyncio.coroutine
+    def stop_chain(self, app):
+        return self.chain.stop()
+
+    def _block_to_dict(self, block):
+        logs_bloom = encode_hex(int_to_big_endian(block.header.bloom))[2:]
+        logs_bloom = '0x' + logs_bloom.rjust(512, '0')
+        return {
+            "difficulty": quantity_encoder(block.header.difficulty),
+            "extraData": encode_hex(block.header.extra_data),
+            "gasLimit": quantity_encoder(block.header.gas_limit),
+            "gasUsed": quantity_encoder(block.header.gas_used),
+            "hash": encode_hex(block.header.hash),
+            "logsBloom": logs_bloom,
+            "mixHash": encode_hex(block.header.mix_hash),
+            "nonce": encode_hex(block.header.nonce),
+            "number": quantity_encoder(block.header.block_number),
+            "parentHash": encode_hex(block.header.parent_hash),
+            "receiptsRoot": encode_hex(block.header.receipt_root),
+            "sha3Uncles": encode_hex(block.header.uncles_hash),
+            "stateRoot": encode_hex(block.header.state_root),
+            "timestamp": quantity_encoder(block.header.timestamp),
+            "totalDifficulty": quantity_encoder(self.chain.chaindb.get_score(block.hash)),
+            "transactions": [encode_hex(tx.hash) for tx in block.transactions],
+            "transactionsRoot": encode_hex(block.header.transaction_root),
+            "uncles": [encode_hex(uncle.hash) for uncle in block.uncles],
+            "size": quantity_encoder(len(rlp.encode(block))),
+            "miner": encode_hex(block.header.coinbase),
+        }
+
+
+class SinglePeerOnDemandDataBackend(OnDemandDataBackend):
+
+    def __init__(self, chaindb):
+        self.chaindb = chaindb
+        self.privkey = ecies.generate_privkey()
+        self._peer = None
+
+    @asyncio.coroutine
+    def get_peer(self):
+        remote = kademlia.Node(NODE_ID, NODE_ADDR)
+        if self._peer is None or self._peer.is_finished:
+            self._peer = yield from asyncio.wait_for(
+                handshake(remote, self.privkey, LESPeer, self.chaindb, NETWORK_ID),
+                HANDSHAKE_TIMEOUT)
+            asyncio.ensure_future(self._peer.start())
+        return self._peer
+
+    @asyncio.coroutine
+    def stop(self):
+        if self._peer is not None and not self._peer.is_finished:
+            yield from self._peer.stop()
+
+
+DemoLightChain = LightChain.configure(
+    'RPCDemoLightChain',
+    vm_configuration=MAINNET_VM_CONFIGURATION,
+    on_demand_data_backend_class=SinglePeerOnDemandDataBackend,
+    network_id=NETWORK_ID,
+)
+
+
+def quantity_encoder(i):
```

Comment:
Can't you just use the built-in `hex` function.  I think it behaves the same as this.

Reply:
Indeed! Changed